client/core/bond.go

```diff
@@ -39,58 +40,217 @@ func (c *Core) watchExpiredBonds(ctx context.Context) {
+//    p = t1 - t0, *variable*, a random process
+//    m = t4 - t3, *variable*, depends on consensus rules and blocks
+//        e.g. several blocks after lockTime passes
+//
+//  - bonds may be spent at t4
+//  - bonds must be replaced by t2 i.e. broadcast by some t2'
+//  - perfectly aligning t4 for bond A with t2' for bond B is impossible on
+//    account of the variable durations
+//  - t2-t2' should be greater than a large percent of expected pending
+//    durations (see pendingBuffer)
+//
+// Here a replacement bond B had a long pending period, and it became active
+// after bond A expired (too late):
+//
+//             t0  t1                         t2' t2                   t3
+//   bond A:   |~~~|--------------------------^---|====================|
+//                                                x
+//   bond B:                                  |~~~~~~|------------------...
+//
+// Here the replacement bond was confirmed quickly, but l was too short,
+// causing it to expire before bond A became spendable:
+//                                                                        > renew as bond C
+//   bond A:   |~~~|----------------------^-------|====================|++‖~~~~~|---...
+//                                          ✓                             x     x
+//   bond B:                              |~|------------------------|=====...
+//
+// Similarly, l could have been long enough to broadcast a replacement in time,
+// but the pending period could be too long (the second "x").
+//
+// Here the replacement bond was broadcast with enough time to confirm before
+// the previous bond expired, and the previous bond became spendable in time to
+// broadcast and confirm another replacement (sustainable):
+//                                                                        > renew as bond C
+//   bond A:   |~~~|----------------------^-------|====================|++‖~~~~~|---...
+//                                               ✓                        ✓     ✓
+//   bond B:                              |~~~~~~|-------------------------------|====...
+//
+// Thus, bond rotation without tier drops requires l>E+m+p. For
+// t3-t0 = p+l+E, this means t3-t0 >= 2(E+p)+m. We will assume the time
+// from authoring to broadcast is negligible, absorbed into the estimate of the
+// max pending duration.
+//
+// tldr:
+//  - when creating a bond, set lockTime = 2*(BondExpiry+pendingBuffer)+spendableDelay
+//  - create a replacement bond at lockTime-BondExpiry-pendingBuffer
+
+// pendingBuffer gives the duration in seconds prior to reaching bond expiry
+// (account, not lockTime) after which a new bond should be posted to avoid
+// account tier falling below target while the replacement bond is pending. The
+// network is a parameter because expected block times are network-dependent,
+// and we require short bond lifetimes on simnet.
+func pendingBuffer(net dex.Network) int64 {
```

Comment:
> I'm still on the fence here, but @chappjc makes a good point. On the timelines we're talking about, we could probably get by without asset-specific values.

Ok, I'll just do it.  Doesn't seem complex, and if we get there with ETH, the `spendableDelay` is likely to be zero (assuming we have a custom contract just for time-locking funds), so that will be helpful.

Reply:
I had started on this Friday but moved on to something else because it was a PITA because I have to get the xcWallets in different spots.  Can work, just requires a bunch of annoying refactoring.